modules/autofill/autofill-api/src/main/java/com/duckduckgo/autofill/store/AutofillStore.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.store
+
+import com.duckduckgo.autofill.domain.app.LoginCredentials
+import kotlinx.coroutines.flow.Flow
+
+interface AutofillStore {
+
+    suspend fun getCredentials(rawUrl: String): List<LoginCredentials>
```

Comment:
For the secure storage, i purposely separated the model containing senstive data (WebsiteLoginCredentials) vs containing non-sensitive data (WebsiteLoginDetails). I think it is a good practice to do so to avoid requesting + keeping in memory sensitive data when we don't really need it.

I think we can use the WebsiteLoginDetails only when:
- showing the bottom sheet for autofill (only shows username)
- showing the view page on the management screen (only shows username + domain)

We then only request the sensitive data WebsiteLoginCredentials when:
- actual autofilling (we need the decrypted password)
- showing the edit page on the management screen (we need the decrypted password)

This is my suggestion but feel free to let me know if you think otherwise.

Reply:
Captured as a task: https://app.asana.com/0/0/1202473050878396/f